include/executors/executor_sycl.hpp

```diff
@@ -257,10 +257,29 @@ class Executor<SYCL> {
    */
   Executor(cl::sycl::queue q) : q_{q} {};
 
-  cl::sycl::queue sycl_queue() {
+  cl::sycl::queue sycl_queue() const {
     return q_;
   }
 
+  enum device_type { UNSUPPORTED_DEVICE, INTELGPU, AMDGPU };
+
+  device_type get_device_type() {
```

Comment:
This is not const, as queue:get_device() is not marked const. Looks like a bug with ComputeCpp.

Reply:
Good catch! we'll fix this soon